In [4]:
pip install statsmodels

Note: you may need to restart the kernel to use updated packages.Requirement already satisfied: statsmodels in c:\users\naufg\appdata\local\programs\python\python311\lib\site-packages (0.14.1)




[notice] A new release of pip is available: 23.1.2 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip


In [5]:
pip install scipy


[notice] A new release of pip is available: 23.1.2 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip


In [6]:
import numpy as np
import pandas as pd
from scipy.stats import norm


# Q1
.Suppose a child psychologist claims that the average time working mothers spend talking to their children is at least 11 minutes per day. You conduct a random sample of 1000 working mothers and find they spend an average of 11.5 minutes per day talking with their children. Assume prior research suggests the population standard deviation is 2.3 minutes.Conduct a test with a level of significance of alpha = 0.05.


In [7]:
#since the sample size is >30 we are using z test here,
#z_test_statistic = (sample_mean - pop_mean) / (sample_std / np.sqrt(sample_size))
#rejection criteria is:z_statistic>critical value , fail to accept Ho

In [8]:
#Ho:pop_mean=11
#H1:pop_mean>11


In [9]:

#Given,

sample_size=1000
sample_mean=11.5
pop_mean=11
pop_std=2.3 
alpha = 0.05 

In [10]:
#calculate the z statistic
z_test_statistic = (sample_mean - pop_mean) / (pop_std / np.sqrt(sample_size))
print('Z-statistic:',z_test_statistic)

Z-statistic: 6.874516652539955


We can calculate the critical Z-value using the norm.ppf() function from scipy.stats

In [11]:
#calculating the critical value for one tail z-test with alpha=0.05
critical_value = norm.ppf(1 - alpha)

print("Critical Z-value:", critical_value)

Critical Z-value: 1.6448536269514722


From the above it is clear that 
Z-statistic > Critical Z-value
therefore , we fail to accept the null hypothesis.
#### Therefore, we have sufficient evidence to conclude that
#### the average time working mothers spend talking to their children is significantly more than 11 minutes per day.

# Q2.
A coffee shop claims that their average wait time for customers is less than 5 minutes. 
To test this claim, a sample of 40 customers is taken, and their wait times are recorded. 
The sample mean wait time is found to be 4.6 minutes with a standard deviation of 0.8 minutes.
Perform a hypothesis test at a significance level of 0.05 and determine whether there is enough evidence to support the coffee shop's claim.

In [12]:
#stating the hypothesis
#Ho:mean>=5
#H1:mean<5


In [13]:
sample_mean=4.6
population_mean=5
sample_size=40
sample_std=0.8
sgnificance_level=0.05

Here the population_std is unknown , So we can use t test for this.

If the sample size is small (typically less than 30) and the population standard deviation is unknown, the t-test is appropriate. However, for larger sample sizes (usually greater than 30), the t-test can also be used even if the population standard deviation is unknown due to the Central Limit Theorem, which states that the distribution of sample means approaches a normal distribution as the sample size increases.

In [14]:
t_test_statistic=(sample_mean - population_mean) / (sample_std / np.sqrt(sample_size))
print('t_statistic:',t_test_statistic)

t_statistic: -3.162277660168382


In [19]:
degrees_of_freedom=sample_size-1
print('df:',degrees_of_freedom)

df: 39


t.ppf(alpha, df) calculates the critical value from the t-distribution corresponding to the given significance level and degrees of freedom. The function ppf stands for "percent point function" and gives you the value such that a random variable from the distribution is less than or equal to that value with the given probability.

In [22]:
#calculating the critical value with df 39 and alpha 0.05
from scipy.stats import t
critical_value = t.ppf(alpha, degrees_of_freedom)

print("Critical value:", critical_value)

Critical value: -1.6848751194974


Since our calculated t-value (-3.16) is less than the critical t-value (-1.685), we fail to accept the null hypothesis.

There is enough evidence to support the claim that the average wait time for customers is less than 5 minutes at a significance level of 0.05.




